### ISSUE TITLE
Broken on 1.33.0

### ISSUE THREAD

**Issue by mkleinbort** [joy13975/streamlit-nested-layout#5]
Hello, fyi this broke on streamlit 1.33.0, I'll add a reproducible example later today.

**Comment by mkleinbort-ic** [comment]
@c-yeh - FYI: 

```python
import streamlit as st 
import streamlit_nested_layout

c1, c2 = st.columns(2)

with c1: 
    c1,c2 = st.columns(2)
    
    with c1:
        st.columns(2)
        
```

Results in:

![image](https://github.com/joy13975/streamlit-nested-layout/assets/112868935/1067c221-4605-49db-b7ff-5b3ad64f0796)

**Comment by joy13975** [comment]
@mkleinbort fixed just now with 0.1.3 ([link](https://pypi.org/project/streamlit-nested-layout/#files)).
Demo works without error.
Please try with `pip install -U streamlit-nested-layout`.

**Comment by mkleinbort-ic** [comment]
I found a fix - simply removing line #21 in `nest.py` is good for my use case.

```python
parent_block_types = list(dg._parent_block_types) # <--- remove this line to fix the issue
```

On that note, I noticed the full package is now basically just:

```python
from typing import cast, Union

from streamlit import cursor
from streamlit.delta_generator import DeltaGenerator
from streamlit.delta_generator import _enqueue_message
from streamlit.elements.form import FormData, current_form_id
from streamlit.proto import Block_pb2, ForwardMsg_pb2
from streamlit.runtime import caching

def _nestable_block(
    self,
    block_proto: Block_pb2.Block = Block_pb2.Block(),
    dg_type: Union[type, None] = None,
) -> "DeltaGenerator":
    dg = self._active_dg

    block_type = block_proto.WhichOneof("type")
   
    if dg._root_container is None or dg._cursor is None:
        return dg

    msg = ForwardMsg_pb2.ForwardMsg()
    msg.metadata.delta_path[:] = dg._cursor.delta_path
    msg.delta.add_block.CopyFrom(block_proto)

    block_cursor = cursor.RunningCursor(
        root_container=dg._root_container,
        parent_path=dg._cursor.parent_path + (dg._cursor.index,),
    )


    if dg_type is None:
        dg_type = DeltaGenerator

    block_dg = cast(
        DeltaGenerator,
        dg_type(
            root_container=dg._root_container,
            cursor=block_cursor,
            parent=dg,
            block_type=block_type,
        ),
    )

    block_dg._form_data = FormData(current_form_id(dg))

    dg._cursor.get_locked_cursor(add_rows_metadata=None)
    _enqueue_message(msg)

    caching.save_block_message(
        block_proto,
        invoked_dg_id=self.id,
        used_dg_id=dg.id,
        returned_dg_id=block_dg.id,
    )

    return block_dg


DeltaGenerator._block = _nestable_block
from typing import cast, Union

from streamlit import cursor
from streamlit.delta_generator import DeltaGenerator
from streamlit.delta_generator import _enqueue_message
from streamlit.elements.form import FormData, current_form_id
from streamlit.proto import Block_pb2, ForwardMsg_pb2
from streamlit.runtime import caching

def _nestable_block(
    self,
    block_proto: Block_pb2.Block = Block_pb2.Block(),
    dg_type: Union[type, None] = None,
) -> "DeltaGenerator":
    dg = self._active_dg

    block_type = block_proto.WhichOneof("type")
   
    if dg._root_container is None or dg._cursor is None:
        return dg

    msg = ForwardMsg_pb2.ForwardMsg()
    msg.metadata.delta_path[:] = dg._cursor.delta_path
    msg.delta.add_block.CopyFrom(block_proto)

    block_cursor = cursor.RunningCursor(
        root_container=dg._root_container,
        parent_path=dg._cursor.parent_path + (dg._cursor.index,),
    )


    if dg_type is None:
        dg_type = DeltaGenerator

    block_dg = cast(
        DeltaGenerator,
        dg_type(
            root_container=dg._root_container,
            cursor=block_cursor,
            parent=dg,
            block_type=block_type,
        ),
    )

    block_dg._form_data = FormData(current_form_id(dg))

    dg._cursor.get_locked_cursor(add_rows_metadata=None)
    _enqueue_message(msg)

    caching.save_block_message(
        block_proto,
        invoked_dg_id=self.id,
        used_dg_id=dg.id,
        returned_dg_id=block_dg.id,
    )

    return block_dg


DeltaGenerator._block = _nestable_block
```

So, pending a fix, a user can just define 

```python

def enable_nesting_of_components():
   '''Modifies the streamlit DeltaGenerator to make almost everything nestable'''
   
   from typing import cast, Union

   from streamlit import cursor
   from streamlit.delta_generator import DeltaGenerator
   from streamlit.delta_generator import _enqueue_message
   from streamlit.elements.form import FormData, current_form_id
   from streamlit.proto import Block_pb2, ForwardMsg_pb2
   from streamlit.runtime import caching

   def _nestable_block(
      self,
      block_proto: Block_pb2.Block = Block_pb2.Block(),
      dg_type: Union[type, None] = None,
   ) -> "DeltaGenerator":
      dg = self._active_dg

      block_type = block_proto.WhichOneof("type")
      
      if dg._root_container is None or dg._cursor is None:
         return dg

      msg = ForwardMsg_pb2.ForwardMsg()
      msg.metadata.delta_path[:] = dg._cursor.delta_path
      msg.delta.add_block.CopyFrom(block_proto)

      block_cursor = cursor.RunningCursor(
         root_container=dg._root_container,
         parent_path=dg._cursor.parent_path + (dg._cursor.index,),
      )


      if dg_type is None:
         dg_type = DeltaGenerator

      block_dg = cast(
         DeltaGenerator,
         dg_type(
               root_container=dg._root_container,
               cursor=block_cursor,
               parent=dg,
               block_type=block_type,
         ),
      )

      block_dg._form_data = FormData(current_form_id(dg))

      dg._cursor.get_locked_cursor(add_rows_metadata=None)
      _enqueue_message(msg)

      caching.save_block_message(
         block_proto,
         invoked_dg_id=self.id,
         used_dg_id=dg.id,
         returned_dg_id=block_dg.id,
      )

      return block_dg

   DeltaGenerator._block = _nestable_block
```

Which can be called by the streamlit app without depending on this package.

**Comment by joy13975** [comment]
@joy13975
Thanks for the analysis - indeed it's a very simple package with very simple fixes.
For sure you're free to just copy the code and DIY.

Please see [comment](https://github.com/joy13975/streamlit-nested-layout/issues/5#issuecomment-2039325040) as package is already fixed.

**Comment by mkleinbort-ic** [comment]
Yes, it's fixed in `streamlit-nested-layout == 0.1.3` - I'll close the issue.
